In [3]:
import sqlite3
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
%matplotlib inline

In [4]:
con = sqlite3.connect('take.db')

In [5]:
tiles = pd.read_sql("select * from cv_piece_raw", con)

In [6]:
tiles = tiles.set_index(["episode_id", "id"])

In [7]:
tiles.head()

r           g           b           h  \
episode_id id                                                        
r2.10      tile-13  143.137517  244.644860   11.174232   42.626836   
           tile-7    11.154438  241.869822  111.375740   72.049704   
           tile-5    11.483669  240.073662  110.711605   71.376650   
           tile-4    11.179123  143.223595  244.788758  102.109327   
           tile-9    11.769231  240.940760  111.261715   71.386384   

                             s           v  orientation        h_skew  \
episode_id id                                                           
r2.10      tile-13  241.875834  245.367156    32.541958  right-skewed   
           tile-7   240.817160  242.338462    22.245955     symmetric   
           tile-5   238.566366  240.884642     6.379068   left-skewed   
           tile-4   241.885732  245.515133    30.315793  right-skewed   
           tile-9   238.613616  242.626879    21.377890  right-skewed   

                           v_skew  num_edges       position       pos_x  \
episode_id id                                                             
r2.10      tile-13      symmetric          6    left center   59.333333   
           tile-7   bottom-skewed          8  center center  141.666667   
           tile-5   bottom-skewed          9    left center  101.666667   
           tile-4       symmetric          6   right center  534.333333   
           tile-9       symmetric          6    left center   99.000000   

                         pos_y  
episode_id id                   
r2.10      tile-13   73.777778  
           tile-7    72.444444  
           tile-5    80.444444  
           tile-4   124.888889  
           tile-9   178.666667

In [13]:
referent = pd.read_sql("select * from referent", con)

In [17]:
tiles_referent = pd.merge(referent, tiles, how="inner", left_on=["object"], right_on=["id"])

In [18]:
tiles_referent

episode_id  object           r           g           b           h  \
0           r6.163  tile-0  244.332317   11.141463  142.926829  161.509146   
1           r6.163  tile-0    5.489344  147.234442  190.470588   92.950554   
2           r6.163  tile-0  244.732645   47.008032  165.239243  160.512909   
3           r6.163  tile-0  218.846397  218.846397  218.846397    0.000000   
4           r6.163  tile-0  142.149639  142.149639  142.149639    0.000000   
...            ...     ...         ...         ...         ...         ...   
1102370      r7.72  tile-3  148.309067  199.565189   29.839179   37.921906   
1102371      r7.72  tile-3  192.149853  206.175221    0.000000   31.501475   
1102372      r7.72  tile-3  134.888812   72.521195  242.076442  130.544823   
1102373      r7.72  tile-3    0.271266  239.277294  143.639652   76.589417   
1102374      r7.72  tile-3  112.336411    0.000000  210.643393  134.426428   

                  s           v  orientation        h_skew         v_skew  \
0        241.765244  244.720732    34.277244  right-skewed      symmetric   
1        237.682012  188.159420   -13.686721   left-skewed      symmetric   
2        206.089501  245.604131   -43.086556  right-skewed     top-skewed   
3          0.000000  217.101770    34.951181     symmetric      symmetric   
4          0.000000  139.140024    -7.071435     symmetric      symmetric   
...             ...         ...          ...           ...            ...   
1102370  210.960291  197.127068    -0.104375     symmetric      symmetric   
1102371  251.088496  204.898525    22.550737     symmetric  bottom-skewed   
1102372  178.371786  243.712995   -16.823584     symmetric      symmetric   
1102373  251.582050  239.456798   -30.164560  right-skewed      symmetric   
1102374  252.057126  209.913445    -6.028783  right-skewed      symmetric   

         num_edges     position       pos_x       pos_y  
0                6    right top  572.000000  326.666667  
1                6     left top  102.000000  332.444444  
2                6  left center   57.666667   71.555556  
3                8    right top  575.666667  429.333333  
4               12  left center  100.666667  125.777778  
...            ...          ...         ...         ...  
1102370         12    right top  500.000000  430.222222  
1102371          8    right top  540.333333  426.222222  
1102372          5    right top  577.000000  381.777778  
1102373          8    right top  578.000000  328.000000  
1102374          8     left top  102.000000  326.666667  

[1102375 rows x 15 columns]

In [19]:
words = pd.read_sql("select * from hand", con)

In [20]:
words.head()

episode_id  inc   word  start_time  end_time
0     r6.163    1   dann        0.07      0.68
1     r6.163    2  <sil>        0.68      1.68
2     r6.163    3  <sil>        1.68      1.78
3     r6.163    4    aus        1.78      2.06
4     r6.163    5    der        2.06      2.24

In [21]:
tiles_referent_words = pd.merge(tiles_referent, words, how="inner", on=["episode_id"])

In [22]:
rot = tiles_referent_words[tiles_referent_words.word == "rot"]
non_rot = tiles_referent_words[tiles_referent_words.word != "rot"]

In [23]:
np.average(rot["r"]), np.average(non_rot["r"])

(128.60296968587485, 128.01841515659487)

In [24]:
links = tiles_referent_words[tiles_referent_words.word == "links"]
non_links = tiles_referent_words[tiles_referent_words.word != "links"]

In [25]:
np.average(links["pos_x"]), np.average(non_links["pos_x"])

(317.89665563397796, 317.8508956845877)

In [28]:
len(set(words["word"]))
#set(words["word"])

383

In [27]:
blue_words = tiles_referent_words[tiles_referent_words.word == "blau"]
non_blue_words = tiles_referent_words[tiles_referent_words.word != "blau"]

np.average(blue_words["b"]), np.average(non_blue_words["b"]) 

(122.2563587916673, 121.70092867402442)

## Questions

-  For the episodes that have the word "rot" in them, what is the average r column value? How does it compare to all of the episodes that don't have the word "rot" in them?
        
              with "rot"         without "rot"
        (128.60296968587485, 128.01841515659487)
        
        The r value averages similar.
        
- What is the average pos_x value for episodes that had the word "links" ? How does it compare to episodes that don't have the word "links" in them?

           with "links"     without "links"
       (317.89665563397796, 317.8508956845877)
       
       The pos_x value averages are similar.
        
- What is the size of the vocabulary? (i.e., what is the set of words in the words dataframe?)

        383
        
- Choose another word in the vocabulary that is not a color. Which column do you think it will correspond to? Following what you did for "rot" and "links", can you see if your assumptions are correct?

        I chose the word "blau" which means blue in german. I would expect that the average value of the `b` column would be similar based on the results from above.
        
             with "blau"       without "blau"
        (122.2563587916673, 121.70092867402442)